packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -519,9 +522,7 @@ export class DatabaseCluster extends DatabaseClusterBase {
         monitoringRoleArn: monitoringRole && monitoringRole.roleArn,
       });
 
-      instance.applyRemovalPolicy(props.removalPolicy, {
-        applyToUpdateReplacePolicy: true,
-      });
+      applyInstanceDeletionPolicy(instance, props.removalPolicy);
 
```

Comment:
It's harder to reason about the code if we modify these properties from underneath the Cfn resources. Instead, can we add it to the constructor properties of `CfnDBInstance`

Something like -

```ts
const instanceDeletionPolicy = computeDeletionPolicy(props.removalPolicy);
const instance = new CfnDBInstance(this, `Instance${instanceIndex}`, {
  ...
  deletionPolicy: instanceDeletionPolicy,
  updateReplicaPolicy: instanceDeletionPolicy,
```

Reply:
That's not really an option in this change - this is all auto-generated code, so we would have to modify `cfn2ts` to allow this.